packages/back-end/src/services/athena.ts

```diff
@@ -1,20 +1,52 @@
-import { Athena } from "aws-sdk";
-import { ResultSet } from "aws-sdk/clients/athena";
+import { STSClient, AssumeRoleCommand } from "@aws-sdk/client-sts";
+import { Athena, ResultSet } from "@aws-sdk/client-athena";
 import { AthenaConnectionParams } from "../../types/integrations/athena";
 import { logger } from "../util/logger";
 import { IS_CLOUD } from "../util/secrets";
 import { ExternalIdCallback, QueryResponse } from "../types/Integration";
 
-function getAthenaInstance(params: AthenaConnectionParams) {
+async function assumeRole(params: AthenaConnectionParams) {
+  // build sts client
+  const client = new STSClient({ region: "us-east-1" });
+  const command = new AssumeRoleCommand({
+    RoleArn: params.assumeRoleARN,
+    RoleSessionName: params.roleSessionName,
+    ExternalId: params.externalId,
+    DurationSeconds: 900,
```

Comment:
It looks like `params.durationSeconds` was added, but is not being used here

Reply:
I'll look at passing that through from the connection.

Reply:
passed from connection